pkg/cmd/codespace/create.go

```diff
@@ -242,12 +244,26 @@ func (a *App) Create(ctx context.Context, opts createOptions) error {
 		}
 	}
 
+	if opts.ssh {
+		opts.showStatus = true
+	}
+
 	if opts.showStatus {
 		if err := a.showStatus(ctx, codespace); err != nil {
 			return fmt.Errorf("show status: %w", err)
 		}
 	}
 
+	if opts.ssh {
+		sshOpts := sshOptions{
+			codespace: codespace.Name,
+		}
+		var sshArgs []string
+		if err := a.sshClient(a, ctx, sshArgs, sshOpts); err != nil {
```

Comment:
This will always happen after `showStatus()`, which (to my understanding) will start its own liveshare session. Since the ssh connection also starts a liveshare connection, would it be feasible to reuse the same connection between the two operations?

Reply:
I think it's feasible, but it may not be simple/clean. This is what I'm seeing:

Get and pass along the session from here:

https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/pkg/cmd/codespace/create.go#L380

https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/internal/codespaces/states.go#L44

https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/internal/codespaces/codespaces.go#L65

And pass it in to `SSH` 

https://github.com/cli/cli/blob/4c237708d878008e85e8190d310dc1826b463ca9/pkg/cmd/codespace/ssh.go#L123

But the SSH command is doing some checks before creating the session in order to provide better errors that we'd miss out on or need to refactor out.

Do you think it's important to reuse the connection or is it more a matter of good-practice? Considering we currently use 2 connections to create and then SSH, this PR doesn't make it any worse, but I don't want to use that as an excuse for bad code. If it isn't critical, can we save it for a follow-up PR? I think we'll also do some refactoring to combine the `codespace` and `codespaces` packages.

Reply:
+1 for leaving this to a second PR, with that said, not too long ago we measured the time it takes to establish a fresh liveshare connection and it is in the ~1-2s range, so it would indeed be a nice perf improvement all around if we could reuse it.